### DISCUSSION TITLE
State pruning

### DISCUSSION THREAD

**Discussion by mrwillis** [0xPolygon/polygon-edge#315] [Ideas]
Matic has an issue right now that with 4-5s block times, they've grown to almost 10TB in archive mode in about a year, with most of it occurring in the past 6-8 months. 

https://snapshots.matic.today/

The SDK has similar constraints/performance and a heavy public chain will definitely reach this number as well.

We propose a feature to add some basic pruning such as what geth 1.10 does based on snapshots.

https://forum.matic.network/t/bor-offline-pruning/1830
https://gist.github.com/yorickdowne/3323759b4cbf2022e191ab058a4276b2
https://blog.ethereum.org/2021/03/03/geth-v1-10-0/

It would be very useful to have some sort of pruning feature as more projects launch with the SDK to guard against this.

**Comment by SwaroopH** [comment]
Can confirm this would be super useful having run Polygon PoS infra since day 1 :) For now, I believe node operators would benefit from "full" nodes too -  if I understand correctly, Polygon Edge run like an archive node?

Also, Geth is planning to add auto-pruning so we wouldn't need to run the 3-5 process of bringing down full nodes every week. Happy to give more feedback as we integrate Edge into our new project.


I do see that geth intends to make pruning by default. for nw
